tests/test_server.py

```diff
@@ -17,117 +41,284 @@ def get_app(self):
+            method='GET',
+            headers={
+                'Accept': 'html'
+            }
+        )
+        self.assertTrue(round_result.code, 200)
+
+        round_result_page = round_result.body.decode()
 
-        self.http_client.fetch(
-            self.get_url('/games/2/board?token={}'.format(self.user.token))
+        self.assertTrue("id: '{}'".format(ID) in round_result_page)
+        self.assertTrue(
+            "score: '{}'".format(move_result['score']) in round_result_page
         )
 
-        self.io_loop.call_later(
-            1,
-            lambda: self.http_client.fetch(
-                self.get_url('/games/2'),
-                callback,
+        deleted = yield self.client.fetch(
+            self.get_url('/games/{}?token={}'.format(ID, TOKEN)),
+            method='DELETE',
+        )
+        self.assertEqual(deleted.code, 200)
+        self.assertEqual(len(server.game_rooms), 0)
+
+    @unittest.mock.patch(
+        'server.game_rooms', {
+            ID: GameRoom(
+                _id=ID,
+                author=LOGIN
             )
+        }
+    )
+    @tornado.testing.gen_test(timeout=100000)
+    def test_games_list(self):
+        result = yield self.client.fetch(
+            self.get_url('/games'),
+            method='GET',
+            headers={
+                'Accept': 'html'
+            }
         )
 
-        self.wait()
-
-    def test_rejoin(self):
-        def callback(response):
-            response = response.body.decode()
-
-            self.assertEqual(response.count('new Player('), 1)
-            self.stop()
+        body_str = result.body.decode()
+        expected = '<a href="/games/{}">'.format(ID)
+        self.assertTrue(expected in body_str)
 
-        self.test_join()
+    @tornado.testing.gen_test(timeout=100000)
+    def test_wrong_game(self):
+        with self.assertRaises(tornado.httpclient.HTTPError):
+            response = yield self.client.fetch(
+                self.get_url('/games/{}'.format(ID)),
+                method='GET'
+            )
+            self.assertEqual(response.code, 404)
 
-        self.http_client.fetch(
-            self.get_url('/games/2/board?token={}'.format(self.user.token))
+    @tornado.testing.gen_test(timeout=100000)
+    def test_oauth_login(self):
+        index_page = yield self.client.fetch(
+            self.get_url('/gh-oauth'),
+            method='GET',
         )
-
-        self.io_loop.call_later(
-            1,
-            lambda: self.http_client.fetch(
-                self.get_url('/games/2'),
-                callback,
-            )
+        root = fromstring(index_page.body.decode())
+        auth_url = root.xpath('//a')[0].attrib['href']
+
+        qs = parse_qs(urlparse(auth_url).query)
+        self.assertTrue('client_id' in qs)
+        self.assertEqual(settings.GH_OAUTH_CLIENT_ID, qs['client_id'][0])
+
+        login_form = yield self.client.fetch(auth_url, method='GET')
```

Comment:
We don't need to test GitHub (send real http request) - we trust that it's tested by GitHub team. So you can just check is `auth_url` pointing to github (https://github.com/login/oauth/access_token).